tests/headers/disable-canonical-naming.hpp

```diff
@@ -0,0 +1,24 @@
+// bindgen-flags: --disable-canonical-naming
+
+struct foo {
```

Comment:
So I think this needs more tests with the interactions with C++ features like:

 * Namespaces.
 * Typedefs.
 * Nested C++ classes.

This surely works on the simple case, but have we figured out what's the desired behavior for all those things?

Do we expect:

```cpp
struct foo {
    typedef int bar;
};
```

To generate a `foo_bar` typedef or a `bar` typedef?

Reply:
I didn't thought of that, thanks. The rationale is: If C++ adds scoping then we should also add it.

In your example, C++ will add scoping, so the type is `foo::bar`. So we should produce `foo_bar`.
I will add more tests and fix these issues.

Reply:
Based on my above comment, maybe we should also change the name of the feature? `disable_canonical_naming` might not be the best option.

Any ideas?